src/queries/role.ts

```diff
@@ -11,20 +13,37 @@ import type {
   QueryByQuery,
 } from './makeQueries';
 
+type Role = AccessibilityRole | TextMatch;
```

Comment:
Currently the effective type of `Role` is `TextMatch` === `string | RegExp`, as `AccessibilityRole` role is string literal union and this gets merged into `string` from `TextMatch`. So I think we should remove `AccessibilityRole` reference here, as it does not change anything but can confuse the reader.

Alternatively, we could allow for only `AccessibilityRole` values, but that would be a breaking change and also would go against RTL API which accepts wide type of `string | RegExp`.

Reply:
Yeah that's something I wanted to discus, forgot to raise in the PR. Another solution is to use [this hack](https://github.com/testing-library/dom-testing-library/blob/main/types/matches.d.ts#L11) to get autocomplete on the name, which I think would be a nice feature.

Reply:
Hmmm, that's an interesting approach. From the thread you linked I conjured this locally:
```ts
type HintString<T extends string> = T | Omit<T, T>;
type Role = HintString<AccessibilityRole> | RegExp;
```

I like the autocompletion but the solution feels a hacky and type information in hover popups is actually harder to understand. All in all, I think we just should go with simple `TextMatch`.